xbmc/guilib/guiinfo/GUIInfoLabel.cpp

```diff
@@ -249,6 +261,12 @@ std::string CGUIInfoLabel::ReplaceAddonStrings(std::string &&label)
   return std::move(label);
 }
 
+std::string CGUIInfoLabel::ReplaceControllerStrings(std::string &&label)
+{
+  ReplaceSpecialKeywordReferences(label, "FEATURE", ControllerFeatureReplacer);
+  return std::move(label);
```

Comment:
```suggestion
  return label;
```
The `std::move` shouldn't be needed here, see [cppreference](https://en.cppreference.com/w/cpp/language/copy_elision):

> In a return statement or a throw-expression, if the compiler cannot perform copy elision but the conditions for copy elision are met, or would be met except that the source is a function parameter, the compiler will attempt to use the move constructor even if the source operand is designated by an lvalue(until C++23) the source operand will be treated as an rvalue(since C++23); see [return statement](https://en.cppreference.com/w/cpp/language/return#Notes) for details.

Reply:
It looks like std::move is required:

```
std::string CGUIInfoLabel::ReplaceControllerStrings(std::string&& label)
{
  ReplaceSpecialKeywordReferences(label, "FEATURE", ControllerFeatureReplacer);
  return label;
         ^
         local variable 'label' will be copied despite being returned by name
}
```

Reply:
Hmm, where is that message coming from?

Reply:
Never mind, the message is correct. Only if the parameter is passed by value the move is done automatically, e.g.
```
str::string test(std::string s)
{
  return std::move(s);
}
```
In this case clang warns about about a redundant move statement. Stupid C++ standard :sob: